tests/test_events_api.py

```diff
@@ -0,0 +1,137 @@
+
+
+class MyTestEvent(BaseEvent):
+    type = "MY_TEST_EVENT"
+
+
+class TestBaseNewEvent:
+    def test_base_new_event(self) -> None:
+        new_event = MyTestEvent({"x": 1})
+
+        assert hasattr(new_event, "raw_data")
+        assert hasattr(new_event, "data")
+        assert new_event.type == "MY_TEST_EVENT"
+        assert new_event.raw_data == {"x": 1}
+        assert new_event.data == '{"x": 1}'
+
+
+class TestEventApi:
+    event_api: Optional[EventAPI] = None
+
+    def setup_method(self) -> None:
+        # pylint: disable= attribute-defined-outside-init
+        with patch.dict(environ, {"AWS_LAMBDA_FUNCTION_NAME": "million-dollar-lambda"}):
+            self.event_api = EventAPI()
+
+    def teardown_method(self, _test_method) -> None:
+        # pylint: disable= attribute-defined-outside-init
+        self.event_api._del()  # pylint: disable=protected-access
+        self.event_api = None
+
+    def test_init_with_defaults(self) -> None:
+        # pylint: disable=protected-access
+        assert self.event_api._pending_events == []
+        assert self.event_api._sent_events == []
+        assert self.event_api._source == "million-dollar-lambda"
+        assert self.event_api._bus_name == "million-dollar-lambda-event-bus"
+        assert self.event_api._resources == []
+
+    def test_set_source(self) -> None:
+        # pylint: disable=protected-access
+        self.event_api.set_source("XXX")
+        assert self.event_api._source == "XXX"
+
+    def test_set_resource(self) -> None:
+        # pylint: disable=protected-access
+        self.event_api.set_resources(["x", "Y"])
+        assert self.event_api._resources == ["x", "Y"]
+
+    def test_register(self) -> None:
+        # pylint: disable=protected-access
+        event = MyTestEvent({"x": 1})
+        self.event_api.register(event)
+
+        assert self.event_api._pending_events == [event]
+
+    def test_get_all_registered_events(self) -> None:
+        event_1 = MyTestEvent({"x": 1})
+        event_2 = MyTestEvent({"x": 2})
+        self.event_api.register(event_1)
+        self.event_api.register(event_2)
+        assert self.event_api.get_all_pending_events() == [event_1, event_2]
+
+    @patch.object(Boto3Client, "eventbridge")
+    @patch.object(EventAPI, "_mark_sent")
+    def test_send(self, mock__mark_sent: MagicMock, mock_send: MagicMock) -> None:
+        event = MyTestEvent({"x": 1})
+        self.event_api.register(event)
+
+        self.event_api.send()
+
+        mock_send.put_events.assert_called_once_with(
+            Entries=[
+                {
+                    "Detail": '{"x": 1}',
+                    "DetailType": "MY_TEST_EVENT",
+                    "EventBusName": "million-dollar-lambda-event-bus",
+                    "Resources": [],
+                    "Source": "million-dollar-lambda",
+                }
+            ]
+        )
+        mock__mark_sent.assert_called_once()
+
+    def test_clear(self):
```

Comment:
Besides the above comments, you should improve test naming. I know tests in `lbz` are not ideal but I don't want to make this situation worse when we write new code.